client/components/PaymentProcessor.js

```diff
@@ -26,33 +34,41 @@ const PaymentProcessor = ({navigation, route}) => {
 
   useEffect(() => {
     setHasErrored(false);
-    fetch(paymentUrl, fetchOptions)
-      .then((response) => {
-        if (response.status < 400) {
-          return response.json();
-        } else {
+    fetch(paymentUrl, PreFetchOptions)
```

Comment:
Same as above (adding a suggestion so you can use github auto apply if you want 😄 )
```suggestion
    fetch(paymentUrl, preFetchOptions)
```